app/components/post_list/post/body/message/show_more_button.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+
+import CompassIcon from '@components/compass_icon';
+import TouchableWithFeedback from '@components/touchable_with_feedback';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type {Theme} from '@mm-redux/types/preferences';
+
+type ShowMoreButtonProps = {
+    highlight: boolean;
+    onPress: () => void;
+    showMore: boolean;
+    theme: Theme;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        button: {
+            flex: 1,
+            flexDirection: 'row',
+        },
+        buttonContainer: {
+            alignItems: 'center',
+            justifyContent: 'center',
+            backgroundColor: theme.centerChannelBg,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            borderRadius: 22,
+            borderWidth: 1,
+            height: 44,
+            width: 44,
+            paddingTop: 7,
+        },
+        container: {
+            alignItems: 'center',
+            justifyContent: 'center',
+            flex: 1,
+            flexDirection: 'row',
+            position: 'relative',
+            top: 10,
+            marginBottom: 10,
+        },
+        dividerLeft: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.2),
+            flex: 1,
+            height: 1,
+            marginRight: 10,
+        },
+        dividerRight: {
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.2),
+            flex: 1,
+            height: 1,
+            marginLeft: 10,
+        },
+        gradient: {
+            flex: 1,
+            height: 50,
+            position: 'absolute',
+            top: -50,
+            width: '100%',
+        },
+        sign: {
+            color: theme.linkColor,
+        },
+    };
+});
+
+const ShowMoreButton = ({highlight, onPress, showMore = true, theme}: ShowMoreButtonProps) => {
+    const style = getStyleSheet(theme);
+
+    let iconName = 'chevron-down';
+    if (!showMore) {
+        iconName = 'chevron-up';
+    }
```

Comment:
Please update `detox/e2e/support/ui/component/post.js`

```
        showLessButton: 'show_more.button.minus',
        showMoreButton: 'show_more.button.plus',
```
to
```
        showLessButton: 'show_more.button.chevron-up',
        showMoreButton: 'show_more.button.chevron-down',
```

I'll fix the associated long post screen e2e tests in a separate PR once this is merged